hathor/manager.py

```diff
@@ -441,13 +441,15 @@ def _initialize_components(self) -> None:
         # self.start_profiler()
         if self._full_verification:
             self.log.debug('reset all metadata')
-            for tx in self.tx_storage.get_all_transactions():
-                tx.reset_metadata()
+            with self.tx_storage.allow_partially_validated_context():
+                for tx in self.tx_storage.get_all_transactions():
```

Comment:
Should `get_all_transactions()` only return transactions in the scope?

Reply:
I think so.

Reply:
Great! Thanks!